### PR TITLE
Fix jeol API docs and privatise module function

### PR DESCRIPTION
**Pull request by ericpre**
FYI @nem1234

### Progress of the PR
- [x] Fix formatting docstring JEOL plugin,
- [x] Privatise functions which shouldnt't part of the public API - since the JEOL reader haven't been released yet, we can privatise these functions without going through a deprecation cycle,
- [n/a] update docstring (if appropriate),
- [n/a] update user guide (if appropriate),
- [n/a] add an changelog entry in the `upcoming_changes` folder (see [`upcoming_changes/README.rst`](https://github.com/hyperspy/hyperspy/blob/RELEASE_next_minor/upcoming_changes/README.rst)),
- [n/a] Check formatting changelog entry in the `readthedocs` doc build of this PR (link in github checks)
- [n/a] add tests,
- [x] ready for review.

### CODE REVIEW

File: hyperspy/io_plugins/jeol.py [hunk 1]
```diff
@@ -191,53 +194,57 @@ def read_img(filename, scale=None, **kwargs):
...
     cutoff_at_kV : float
-	cutoff energy to reduce memory size of spectrum image, default: None (do not cutoff)
-    downsample : int or [int, int]
-    	downsample along spacial axes to reduce memory size of spectrum image
-	value must be 2^n.
-	default: 1 (do not downsample)
+        The maximum energy. Useful to reduce memory size of spectrum image.
+        Default is None (no cutoff)
+    downsample : int or (int, int)
+    	Downsample along spacial axes to reduce memory size of spectrum image
```
Comment (jlaehne) [inline hyperspy/io_plugins/jeol.py]:
```suggestion
    	Downsample along spacial axes to reduce memory size of spectrum image.
        Value must be 2^n. Default is 1 (no downsampling).
```

File: hyperspy/io_plugins/jeol.py [hunk 2]
```diff
@@ -191,53 +194,57 @@ def read_img(filename, scale=None, **kwargs):
...
-    	list of frame numbers to be read (None for all data)
-    frame_shifts : list of [int dy,int dx] or list of [int dy, int dx, int dEn], default None
-    	each frame will be loaded with offset of dy, dx, (and optionary
-        energy axis). Units are pixels/channels.
-        This is useful for express drift correction. Not suitable for accurate analysis.
-        like the result of estimate_shift2D(), the first parameter is for y-axis
-    frame_start_index: list of offset pointer of each frame in the raw data.
+    	List of frame numbers to be read (None for all data)
+    frame_shifts : list of [int, int] or list of [int, int, int], default None
+    	Each frame will be loaded with offset of dy, dx, (and optionary energy
```
Comment (jlaehne) [inline hyperspy/io_plugins/jeol.py]:
```suggestion
    	Each frame will be loaded with offset of dy, dx, (and optional energy
```

File: hyperspy/io_plugins/jeol.py [hunk 3]
```diff
@@ -191,53 +194,57 @@ def read_img(filename, scale=None, **kwargs):
...
+        axis). Units are pixels/channels.
+        This is useful for express drift correction. Not suitable for accurate
+        analysis.
+        Like the result of estimate_shift2D(), the first parameter is for y-axis
+    frame_start_index: list
+        The list of offset pointer of each frame in the raw data.
```
Comment (jlaehne) [inline hyperspy/io_plugins/jeol.py]:
```suggestion
        The list of offset pointers of each frame in the raw data.
```

File: hyperspy/io_plugins/jeol.py [hunk 4]
```diff
@@ -594,50 +601,58 @@ def parsejeol(fd):
...
+        The indices of each frame start. If length is zero, the indices will be
+        determined from rawdata.
+    frame_list : list
+        List of frames to be read.
+    width, height : int
+    	The navigation dimension.
+    channel_number : int
+        The number of channels.
+    width_norm, height_norm : int
+        Rebin factor of the navigation dimension
```
Comment (jlaehne) [inline hyperspy/io_plugins/jeol.py]:
```suggestion
        Rebin factor of the navigation dimension.
```

File: hyperspy/io_plugins/jeol.py [hunk 5]
```diff
@@ -594,50 +601,58 @@ def parsejeol(fd):
...
+    rebin_energy : int
+        Rebin factor of the energy dimension
```
Comment (jlaehne) [inline hyperspy/io_plugins/jeol.py]:
```suggestion
        Rebin factor of the energy dimension.
```

File: hyperspy/io_plugins/jeol.py [hunk 6]
```diff
@@ -594,50 +601,58 @@ def parsejeol(fd):
...
+    frame_shifts : list
+        The list of image positions [[x0,y0,z0], ...]. The x, y, z values can
+        be negative. The data points outside data cube are ignored.
 
+    Returns
+    -------
+    data : numpy.ndarray or dask.array
+        The spectrum image with shape (frame, x, y, energy) if sum_frames is
+        False, otherwise (x, y, energy).
+        If lazy is True, the dask array is a COO sparse array
```
Comment (jlaehne) [inline hyperspy/io_plugins/jeol.py]:
```suggestion
        If lazy is True, the dask array is a COO sparse array.
```

File: hyperspy/io_plugins/jeol.py [hunk 7]
```diff
@@ -594,50 +601,58 @@ def parsejeol(fd):
...
+        The SEM/STEM image with shape (frame, x, y) if sum_frames is False,
+        otherwise (x, y).
+    has_em_image : bool
+        True if the stream contains SEM/STEM images.
+    sweep : int
+	    The number of loaded frames.
+    frame_start_index : list
+        The indices of each frame start. 
+    max_shift : numpy.ndarray
+	    The maximum shifts of the origin in the navigation dimension
```
Comment (jlaehne) [inline hyperspy/io_plugins/jeol.py]:
````suggestion
        The maximum shifts of the origin in the navigation dimension.
```

**Review by jlaehne — APPROVED** [review]

**Comment by nem1234** [comment]
Line number 220 seems broken.
Should be removed.

**Comment by jlaehne** [comment]
> Line number 220 seems broken. Should be removed.

Can you be more specific. As this is already merged, we would have to submit a separate fix.

**Pull request merged**